sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Rules/CheckFileLicenseTest.cs

```diff
@@ -67,135 +71,131 @@ public class CheckFileLicenseTest
-            {
-                // Putting the expected issue within the file make the file being no longer empty so we need to catch the expected issue
-                // from here.
-            }
+            Verifier.VerifyAnalyzer(@"TestCases\CheckFileLicense_MultiSingleLineLicenseStartWithNamespace.cs",
+                new CheckFileLicense { HeaderFormat = MultiSingleLineCommentHeader });
         }
 
         [TestMethod]
         [TestCategory("Rule")]
-        public void CheckEmptyHeaderFormat()
+        public void CheckFileLicense_WhenLicensedWithMultiSingleLineCommentStartingWithNamespaceAndRegex_ShouldBeCompliant()
+        {
+            Verifier.VerifyAnalyzer(@"TestCases\CheckFileLicense_MultiSingleLineLicenseStartWithNamespace.cs",
+                new CheckFileLicense { HeaderFormat = MultiSingleLineRegexHeader, IsRegularExpression = true });
+        }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void CheckFileLicense_WhenEmptyFile_ShouldBeNonCompliant()
         {
-            const string expectedErrorMessage =
-                "Expected collection to be empty, but found {error AD0001: The Compiler Analyzer 'SonarAnalyzer.Rules.CSharp.CheckFileLicense' " +
-                "threw an exception of type 'System.ArgumentException' with message 'Expects a non-empty license header" +
-                "\r\nParameter name: headerFormat'.}.";
             try
             {
-                Verifier.VerifyAnalyzer(@"TestCases\CheckFileLicense_NoLicenseStartWithUsing.cs",
-                    new CheckFileLicense { HeaderFormat = string.Empty });
+                Verifier.VerifyAnalyzer(@"TestCases\CheckFileLicense_EmptyFile.cs",
+                    new CheckFileLicense { HeaderFormat = SingleLineHeader });
             }
-            catch (AssertFailedException ex)
+            catch (AssertFailedException ex) when (ex.Message == "Issue with message 'Add or update the header of this file.' not expected on line 1")
             {
-                if (ex.Message != expectedErrorMessage)
-                {
-                    throw;
-                }
+                // Putting the expected issue within the file make the file being no longer empty so we need to catch the expected issue
+                // from here.
             }
         }
 
         [TestMethod]
         [TestCategory("Rule")]
-        public void CheckInvalidRegex()
+        public void CheckFileLicense_WhenProvidingAnInvalidRegex_ShouldThrowException()
         {
             const string expectedErrorMessage =
                 "Expected collection to be empty, but found {error AD0001: The Compiler Analyzer 'SonarAnalyzer.Rules.CSharp.CheckFileLicense' " +
                 "threw an exception of type 'System.ArgumentException' with message 'Invalid regular expression: " +
                 FailingSingleLineRegexHeader +
                 "\r\nParameter name: headerFormat'.}.";
+            bool hasThrown = false;

```

Comment:
I usually prefer this syntax, which is a bit shorter:

Action action = () => Verifier.VerifyAnalyzer(@"file.cs", new AnalyzerClass());
action.ShouldThrow<AssertFailedException>();